### PR TITLE
fix for resampled stereo audio array size

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

File: libraries/audio/src/Sound.cpp [hunk 1]
```diff
@@ -102,12 +102,11 @@ void Sound::downSample(const QByteArray& rawAudioByteArray) {
     int numSourceSamples = rawAudioByteArray.size() / sizeof(int16_t);
     
     int numDestinationBytes = rawAudioByteArray.size() / 2;
```
Comment (Atlante45) [inline libraries/audio/src/Sound.cpp]: You might want to use sizeof(int16_t) here too.
Or even sizeof(AudioSample).

File: libraries/audio/src/AudioInjectorLocalBuffer.cpp [hunk 2]
```diff
@@ -80,7 +82,7 @@ qint64 AudioInjectorLocalBuffer::recursiveReadFromFront(char* data, qint64 maxSi
     
     // check if we need to call ourselves again and pull from the front again
     if (bytesRead < maxSize) {
-        return bytesRead + recursiveReadFromFront(data, maxSize);
+        return bytesRead + recursiveReadFromFront(data, maxSize - bytesRead);
```
Comment (Atlante45) [inline libraries/audio/src/AudioInjectorLocalBuffer.cpp]: I think that's what you want:
        return bytesRead + recursiveReadFromFront(data + bytesRead, maxSize - bytesRead);

**Comment by hifi-gustavo** [comment]
Build console output for commit: 164c06a1904d6b52c33f1140a7f307a28473a43f available at: https://gist.github.com/c348ce153f69918067db<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4585.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4585.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3f079e10aa1c14444718271864d1a7fe29d1b21f available at: https://gist.github.com/c348ce153f69918067db<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4585.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4585.dmg

**Pull request merged**